flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/consumer/TestInputChannel.java

```diff
@@ -190,6 +190,9 @@ public void resumeConsumption() {
         isBlocked = false;
     }
 
+    @Override
+    public void acknowledgeAllRecordsProcessed() throws IOException {}
```

Comment:
Should we rather call:
```
    @Override
    public void acknowledgeAllRecordsProcessed() throws IOException {
         inputGate.acknowledgeAllRecordsProcessed(this);
    }
```

Reply:
It seems we could not call `inputGate.acknowledgeAllRecordsProcessed(this);` here since the call is from `SingleInputGate` to `InputChannel`, thus we could not call the method in `InputGate` again.

Reply:
You're right? Would it work with an `UnsupportedOperationException`? If not then let's leave it empty.

Reply:
Currently we should be able to throw `UnsupportedOperationException` and not affect the existing tests. I modified all the related  testing `InputChannel` and `InputGate` to throw `UnsupportedOperationException` for now.